### PR TITLE
Implements pin check on learn-only devices

### LINKED ISSUE #9796: Facility Settings Updates: PIN usage on Learn-only Devices
## Overview 

This issue is a full-stack issue that updates allows the Device page to be accessed on learn-only devices by entering a PIN.

We will be using a feature branch for this project `update-facility-settings`. Please make PRs to this branch, which will later be merged into develop when the feature is complete.

### Figma 

When the user picks "Device" on the menu, the modal opens and prompts for the PIN to be entered. If the correct PIN is entered, the user should be authorized to access the device page. 

Screen Shot 2022-10-24 at 11 53 02 AM

**Please see [complete figma spec](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=1473%3A16838) for details on states for PIN modal**

➡️ Accessibility notes: 
* Please use `KComponents` for wherever possible to ensure that the content is properly passed on to screen readers, and that focus outlining and focus shift works. (See [KDS docs](https://kolibri-design-system.netlify.app/) for more information)

### API changes 

Authentication will need to be updated to support a PIN granting access as an admin. A user should not be able to navigate directly via a URL without being prompted to authenticate, but should not have to enter the PIN twice, once when pressing "Device" in the navigation, and then again when redirected. One approach would be to save something in the session to say that it is authenticated, i.e. a 15 second cookie that can be used if the PIN has already been entered, and can be checked during navigation to the Device plugin.

Access should expire on return to Learn plugin and/or with a timeout.

## Acceptance Criteria
- [ ]  Frontend is updated with correct strings, using `KComponents`
- [ ] PIN authentication is implemented as described above
- [ ]  The `FacilitySettingsPage.spec.js` should be updated so that existing tests still pass (as necessary) and there should be at least one test written to cover this scenario

## References 
- [Notion project page](https://www.notion.so/learningequality/Updates-to-Facility-Settings-f35f3be6e5fb45fb8de0abc0d898c017)
- [Figma spec](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=969%3A16669)
- [Gherkin scenarios](https://www.notion.so/learningequality/Gherkin-scenarios-02096ac9925844f8902c818cab607d4b#1959b753421648308fb467e21bdaa125)

### PR DESCRIPTION
**Pull request by akolson**
## Summary

This PR implements a pin check on learn-only devices to restrict access to the device plugin in kolibri

## References

Closes #9796 

## Reviewer guidance

See https://github.com/learningequality/kolibri/issues/9796 for reviewer guidance
To set up pin, see https://github.com/learningequality/kolibri/issues/9789

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/device/assets/src/views/DeviceIndex.vue [hunk 1]
```diff
@@ -8,6 +8,14 @@
       <DeviceTopNav />
     </template>
 
+    <transition name="delay-entry">
```
Comment (rtibbles) [inline kolibri/plugins/device/assets/src/views/DeviceIndex.vue:11]: I think the designs call for this to be shown when clicking on the Device nav item, prior to navigation to the Device page, so we may need to include this in the DeviceSideNav also, and then persist the authentication state across the page reload with a cookie.

(Although, this still _also_ needs to be here to allow for authentication with the PIN if someone navigates directly to the page).

Comment (akolson) [reply kolibri/plugins/device/assets/src/views/DeviceIndex.vue:11]: Thanks @rtibbles! Yes, you are right about this. I did notice this during visual testing. Thanks for the hint!

**Comment by marcellamaki** [comment]
Hi @akolson - I've read through the code and started doing QA on this. it looks good so far and close to done, although I am running into one error with the pin input, despite the correct values being stored. I will write up my comments first thing tomorrow when I have a more clear mind :)

**Comment by marcellamaki** [comment]
Hi @akolson - my follow up comments with a morning brain :) I'll explain my manual QA testing scenario here: 

1. I've added the pin for the facility, and I have confirmed it is saved. 
2. I updated the plugins files so that subset of users device would return true for the `plugin_data`
3. I sign in as a learner, and update the URL manually to `en/device` (separately, I think ensuring that the `Device` option appears for LOD navigation is part of @nucleogenesis's work on the setup wizard, but just making sure I flag here as well so it's not a case of each thinking the other was resolving it in their work)
4. I have console logged the existing pin, which is available in the frontend, so that's good ✅ 
5. I enter the pin as the learner, and get a 403 error. 

My initial guess is that maybe this is because a learner is accessing the `/device` endpoint plugins? I'm not sure how much of this has to do with other setup configurations (see above re: setup wizard) and how much of this requires updates to this PR... not sure if anyone else has an idea here (@rtibbles?) 

https://user-images.githubusercontent.com/17235236/219419199-f537dec6-76e7-4bf8-b84f-7af4f856d7d3.mp4

**Comment by akolson** [comment]
@marcellamaki  Thanks for this early morning brain comments :). I think this is a permission related issue regarding access to the is_pin_valid() resource. We will need to make changes to the backend to allow the learner to access this resource. I think the issue is related to https://github.com/learningequality/kolibri/pull/10073. I however think, it should be okay to fix this here as it directly ties into the functionality of this PR. @rtibbles ?

**Comment by rtibbles** [comment]
I think the correct resolution for this is slightly different. If I am understanding correctly, @marcellamaki is attempting to access the Device page using a learner account that does not have superadmin permissions. This user should just be told that they do not have permission to access this page, as was previously the case. This is because the PIN feature does not grant any additional permissions to users, but rather restricts access to the Device page for users who have superadmin permissions, but are not facility admins.

So, the resolution here is to only do any of the PIN modal/PIN access checks in the case that the user has SUPERADMIN permissions but is not also a facility admin.

**Comment by akolson** [comment]
Ahh I see, let me make this change. Thanks @rtibbles

**Comment by akolson** [comment]
@rtibbles @marcellamaki  also just to clarify, Is this a check that should only happen on learner only devices or any device?

**Comment by rtibbles** [comment]
Oh right - yes, this is targeted at LODs specifically!

**Comment by akolson** [comment]
@marcellamaki  a validation check has been added to ensure that the pin will only be required for superusers who are not facility admins.

**Review by marcellamaki — APPROVED** [review]
Looks good @akolson. I think it's okay to make any further adjustments as we discover them through some robust QA scenarios and bug bashing :) 

Also okay to merge despite mac build being broken!

**Pull request merged**